components/Parallax/Parallax.tsx

```diff
@@ -18,41 +19,22 @@ type ParallaxProps = {
   children: ReactNode;
 };
 
-/**
- * Using guide from https://samuelkraft.com/blog/spring-parallax-framer-motion-guide
- */
-export const Parallax = ({ children, offset = 60 }: ParallaxProps) => {
+export const Parallax = ({ children, offset }: ParallaxProps) => {
   const prefersReducedMotion = useReducedMotion();
-  const { scrollY } = useScroll();
-  const [elementTop, setElementTop] = useState(0);
-  const [clientHeight, setClientHeight] = useState(0);
   const ref = useRef<HTMLDivElement>(null);
+  const { scrollYProgress } = useScroll({
+    target: ref,
+    // This mean aniimation starts when the top of the image touches the bottom of the viewport,
+    // and ends when the bottom of the image touches the top of the viewport.
+    offset: ['start end', 'end start'],
+  });
+  const yRange = useTransform(scrollYProgress, [0, 1], [offset, -offset]);
+  const y = useSpring(yRange, { stiffness: 200, damping: 30 });
```

Comment:
Clean up all the old parallax code and replace with just this - works better and a lot less code